client/rest-high-level/src/main/java/org/elasticsearch/client/security/PutRoleMappingResponse.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.security;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+
+/**
+ * Response when adding/updating a role mapping. Returns a boolean field for
+ * whether the role mapping was created or updated.
+ */
+public final class PutRoleMappingResponse {
+
+    private boolean created;
+
+    public PutRoleMappingResponse(boolean created) {
+        this.created = created;
+    }
+
+    public boolean isCreated() {
+        return created;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final PutRoleMappingResponse that = (PutRoleMappingResponse) o;
+        return created == that.created;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(created);
+    }
+
+    private static final ConstructingObjectParser<PutRoleMappingResponse, Void> PARSER = new ConstructingObjectParser<>(
+            "put_role_mapping_response", true, args -> new PutRoleMappingResponse((boolean) args[0]));
+    static {
+        PARSER.declareBoolean(constructorArg(), new ParseField("created"));
+        PARSER.declareObject((a,b) -> {}, (parser, context) -> null, new ParseField("role_mapping")); // ignore the role_mapping field!
```

Comment:
is there a change here that is missing to move created as a top level field in the PutRoleMappingResponse?

Reply:
I am confused how this works when created is only within role mapping but we ignore role mapping

Reply:
I think I understand what is happening. The parser finds the role_mapping object, and calls our declared methods where we return null and do not actually consume content using the parser. The object parser then moves to the next item which is the created field, and parses it.

I personally do not like this behavior as it is very sneaky. @nik9000 I see your name on a lot of the lines of these classes, maybe you have some thoughts on this behavior?

>  saw we did similar in PutUserResponse

PutUserResponse did this as a BWC method. I added a top-level created field in 6.x so we could remove the nested field in master for 7.0, see https://github.com/elastic/elasticsearch/pull/33512

Reply:
@bizybot can you open up a issue that describes this behavior of the object parser and label it with discuss? Then we can move this PR forward.

Reply:
can you please document how this code works and include a link to that issue in the comment. Its not straightforward and anything like this should be documented